### ISSUE TITLE
bug(share/discovery): discovey should not retry lookup if it has found enough peers

### ISSUE THREAD

**Issue by walldiss** [celestiaorg/celestia-node#2890]
### Implementation ideas

Discovery retries lookup, even when enough peers are already found. Instead it should not start new discovery until one of connected peers disconects. Not critical, just runs discovery lookup, when there is not need for it.

**Comment by Wondertan** [comment]
Duplicate of https://github.com/celestiaorg/celestia-node/issues/2703
